corehq/apps/es/users.py

```diff
@@ -221,48 +221,32 @@ def is_active(active=True):
     return filters.term("is_active", active)
 
 
-def user_data(key, value):
+def _user_data(key, filter_):
+    # Note: user data does not exist in ES for web users
     return queries.nested(
         'user_data_es',
         filters.AND(
             filters.term(field='user_data_es.key', value=key),
-            queries.match(field='user_data_es.value', search_string=value),
+            filter_
         )
     )
 
 
-def _missing_user_data_property(property_name):
-    """
-    A user_data property doesn't exist.
-    """
-    return filters.NOT(
-        queries.nested(
-            'user_data_es',
-            filters.term(field='user_data_es.key', value=property_name),
-        )
-    )
+def query_user_data(key, value):
+    return _user_data(key, queries.match(field='user_data_es.value', search_string=value))
 
 
-def _missing_user_data_value(property_name):
-    """
-    A user_data property exists but has an empty string value.
-    """
-    return queries.nested(
-        'user_data_es',
-        filters.AND(
-            filters.term('user_data_es.key', property_name),
-            filters.NOT(
-                filters.wildcard(field='user_data_es.value', value='*')
-            )
-        )
-    )
+def login_as_user(value):
+    return _user_data('login_as_user', filters.term('user_data_es.value', value))
 
 
 def missing_or_empty_user_data_property(property_name):
     """
     A user_data property doesn't exist, or does exist but has an empty string value.
     """
-    return filters.OR(
-        _missing_user_data_property(property_name),
-        _missing_user_data_value(property_name),
-    )
+    missing_property = filters.NOT(queries.nested(
```

Comment:
This is the only place that uses `filters.wildcard`. Since this has been removed now, would it make sense to remove the wildcard filter entirely?

Reply:
I'm leaning more towards the latter option. We might end up using this again in the future, but then again we might not in which case this just ends up being dead code we have to maintain. Given it's only a couple of lines, we can always add it back if we need it again. But if not, at least we've managed to clean up the codebase a tiny bit.